### PR TITLE
Stack drawer hierarchy #9

### PR DESCRIPTION
**Pull request by ntwigs**
### TL;DR
* Navigation Hierarchy :tokyo_tower: 
* Replacing menu with back-arrow in stack navigation :mahjong: 
    * Only necessary since the drawer menu is placed to the left :fist_left: 
* Stack has been moved to its' own navigation :owl: 
***
### Hierarchy
* The hierarchy of navigation now works like this:
* Switch
    * Drawer / Tab
        * Stack
            * Component
            * Component
        * Stack
             * Component

The switch checks if you're logged in, and if you are, a Drawer or Tab navigator will be selected. The Drawer and Tab navigators are the top navigation. Inside the top navigational components, you have stacks. These stacks are the sub navigational components. :classical_building: 
*** 
### Replacing menu with back-arrow
The drawer opener on Android is placed on the left side in this boilerplate (because material design says so). This leads to a problem when entering the second level of a stack navigation since it did display a menu and not the back button. Therefore it now checks if there is a back (by checking if the previous navigation had a title), and if there is, it displays a back-arrow instead. :100: 
***
### Stack has been moved
It was placed in a logged-in-screen, whereas all the other navigation were placed inside the navigational folder. Therefore I removed the logged-in-screen and placed all stack navigations inside the `/navigators/stack-navigator.tsx`. So now all navigators are in the same place, and all the screens are in the same place. :lemon:

### CODE REVIEW

**Comment by ntwigs** [comment]
It now also works on iOS! 🎉

File: src/navigators/root-navigator.ts [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
 import {SignedOut} from '../screens/signed-out-screen'
 import {Routes} from '../entities'
+import {RootTabs} from './tab-navigator'
+import {Platform} from 'react-native'
+import {RootDrawer} from './drawer-navigator'
+
+/**
+ * Selects tabs for ios and drawer for android
+ */
+const SelectPlatform = Platform.select({
```
[the remark this answers is not in the document]
Comment (SheriefBadran) [reply src/navigators/root-navigator.ts]: Either of us can change that next pr
